### PR TITLE
new localStorage util for managing key storage in the paywall

### PR DESCRIPTION
**Pull request by cellog**
# Description

This adds a new utility library for handling localStorage. Specific functionality is detecting the existence of localStorage and then storing keys.

Keys are indexed by user, network, and lock.

This is a step along the road to #1314 and #1287 

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/utils/localStorage.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+}
+
+export function getUserStorageName(account, network) {
+  return `unlock-${account}-${network}`
+}
+
+export function saveKeyToStorage(window, key, network) {
+  const storageKeys = getKeysFromStorage(window, key.owner, network)
+  const keys = storageKeys ? storageKeys : {}
```
Comment (julien51) [inline unlock-app/src/utils/localStorage.js]: isn't `getKeysFromStorage` returning `{}` by default? if that's the case maybe do 
```
const keys = getKeysFromStorage(window, key.owner, network) 

```

File: unlock-app/src/utils/localStorage.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      storage.length !== 0
+    )
+  }
+}
+
+export function getUserStorageName(account, network) {
+  return `unlock-${account}-${network}`
+}
+
+export function saveKeyToStorage(window, key, network) {
```
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/utils/localStorage.js]: One very helpful thing would to be show the structure of the data being stored (I know it is a string... but the `JSON.parsed` version of it)

**Comment by cellog** [comment]
closing for now, will make a new one on the same branch after massive refactor

**Comment by cellog** [comment]
Hey @julien51 it occurs to me - we don't need this at all if we aren't storing the keys. We already have the account :). So consider this obsoleted until needed. I guess I'll keep the branch around just in case

**Pull request closed without merging**